crates/storage_impl/src/errors.rs

```diff
@@ -140,6 +143,12 @@ impl From<error_stack::Report<DatabaseError>> for StorageError {
     }
 }
 
+impl From<diesel::result::Error> for StorageError {
+    fn from(err: diesel::result::Error) -> Self {
+        StorageError::TransactionFailed(err.into())
+    }
+}
```

Comment:
We can't do `From<diesel::result::Error>` for StorageError since it includes errors from Redis. Ideally we should have `From<diesel::result::Error>` for DatabaseError and we have `From<error_stack::Report<DatabaseError>>` for StorageError

Reply:
>We can't do From<diesel::result::Error> for StorageError since it includes errors from Redis

I didn't get this

Reply:
So we should ideally convert DieselError to DatabaseError and DatabaseError to StorageError

Reply:
So I need to move the TransactionFailed error to DatabaseError, right?